### ISSUE TITLE
`typevar-name-mismatch` check suggestion

### ISSUE THREAD

**Issue by jpy-git** [PyCQA/pylint#6011]
### Current problem

Currently we don't have a check to flag if a user doesn't assign a `TypeVar` to a variable of the same name:
i.e.
```python
T = TypeVar('A')
```
should be
```python
A = TypeVar('A')
```

### Desired solution

There should be a lint along the lines of `TypeVar must be assigned to a variable named "A" (typevar-name-mismatch)`

### Additional context

Pylance currently flags this, but would be nice to also have in Pylint for CI purposes 😄 
<img width="462" alt="image" src="https://user-images.githubusercontent.com/80432516/160381331-7e58f9b7-514e-41c2-82f2-0c41c8c8d606.png">

**Comment by DanielNoord** [comment]
Duplicate of #5224 although I do like this name suggestion. Could you post it in that issue?

**Comment by jpy-git** [comment]
Ah apologies, will do! Happy to take a look at adding this as well 😄
